### PR TITLE
make : Add alternative mirror.

### PR DESCRIPTION
**Pull request by p-groarke**
Specify library name and version:  **make/\***

https://ftpmirror.gnu.org is down, preventing the build of libraries dependent on `make`.

As per loosely this thread https://github.com/conan-io/conan/issues/6349 and this pr https://github.com/conan-io/conan-center-index/pull/632, add a mirror.

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/make/all/conandata.yml [hunk 1]
```diff
@@ -1,10 +1,14 @@
 sources:
   "4.3":
-    url: "http://ftpmirror.gnu.org/gnu/make/make-4.3.tar.gz"
-    sha256: "e05fdde47c5f7ca45cb697e973894ff4f5d79e13b750ed57d7b66d8defc78e19"
+    - url: "https://ftpmirror.gnu.org/gnu/make/make-4.3.tar.gz"
```
Comment (p-groarke) [inline recipes/make/all/conandata.yml:3]: https seems like the better choice

File: recipes/make/all/conandata.yml [hunk 2]
```diff
@@ -1,10 +1,14 @@
...
+      sha256: "e05fdde47c5f7ca45cb697e973894ff4f5d79e13b750ed57d7b66d8defc78e19"
+    - url: "https://ftp.gnu.org/gnu/make/make-4.3.tar.gz"
+      sha256: "e05fdde47c5f7ca45cb697e973894ff4f5d79e13b750ed57d7b66d8defc78e19"
```
Comment (p-groarke) [inline recipes/make/all/conandata.yml:6]: same file same sha

File: recipes/make/all/conanfile.py [hunk 3]
```diff
@@ -38,8 +39,19 @@ def package_id(self):
...
     def source(self):
-        get(self, **self.conan_data["sources"][self.version],
-            destination=self.source_folder, strip_root=True)
+        for data in self.conan_data["sources"][self.version]:
+            url, sha = data.values()
+            filename = url[url.rfind("/")+1:]
+            try:
+                download(self, url, filename, sha256=sha, retry=0)
+            except ConanException as e:
+                if "NewConnectionError" in str(e):
```
Comment (p-groarke) [inline recipes/make/all/conanfile.py:48]: I don't know how / if you can filter exceptions in python so I used this. Better than nothing, lmk if there's a better way.

File: recipes/make/all/conandata.yml [hunk 4]
```diff
@@ -1,12 +1,20 @@
...
-    url: "http://ftpmirror.gnu.org/gnu/make/make-4.2.1.tar.bz2"
-    sha256: "d6e262bf3601b42d2b1e4ef8310029e1dcf20083c5446b4b7aa67081fdffc589"
+    - url: "https://ftpmirror.gnu.org/gnu/make/make-4.2.1.tar.bz2"
+      sha256: "d6e262bf3601b42d2b1e4ef8310029e1dcf20083c5446b4b7aa67081fdffc589"
+    - url: "https://ftp.gnu.org/gnu/make/make-4.2.1.tar.bz2"
+      sha256: "d6e262bf3601b42d2b1e4ef8310029e1dcf20083c5446b4b7aa67081fdffc589"
 patches:
   "4.3":
     - patch_file: "patches/4.3-0001-clang.patch"
+      patch_description: "Fix __alloca define check."
```
Comment (p-groarke) [inline recipes/make/all/conandata.yml:15]: Suppress schema warning to the best of my abilities.

File: recipes/make/all/conanfile.py (hunk 3 above)
Comment (RubenRBS) [reply recipes/make/all/conanfile.py:49]: There's no need for this. The `get`/`download` functions should already be able to handle mirrors

**Review by RubenRBS — CHANGES_REQUESTED** [review]
Thanks a lot!
The mirror URLs are ok, but the changes to the recipes seem unnecessary :)

File: recipes/make/all/conandata.yml (hunk 4 above)
Comment (RubenRBS) [reply recipes/make/all/conandata.yml:15]: Thanks! We don't enforce this for PRs that do not directly add the patch, but it's unlucky that the linter still complains for every other case

File: recipes/make/all/conanfile.py (hunk 3 above)
Comment (RubenRBS) [reply recipes/make/all/conanfile.py:49]: Or does this come from the current get implementation not handling this failing mode properly?

Comment (p-groarke) [reply recipes/make/all/conanfile.py:49]: @RubenRBS Could you point to a recent example? All my research seemed to indicate mirrors coming from conandata.yml needed some custom logic in the recipe, but a lot of the info was old (google is bad nowadayts). If it doesn't that's great :)

Comment (p-groarke) [reply recipes/make/all/conanfile.py:49]: Roger :] I'll ping you re-review when I have a some time to teak

**Review by danimtb — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by jwillikers — APPROVED** [review]

**Pull request closed without merging**